bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/internal/AudioServlet.java

```diff
@@ -36,47 +38,39 @@
 import org.openhab.core.audio.AudioHTTPServer;
 import org.openhab.core.audio.AudioStream;
 import org.openhab.core.audio.FixedLengthAudioStream;
-import org.openhab.core.io.http.servlet.OpenHABServlet;
-import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Deactivate;
-import org.osgi.service.component.annotations.Reference;
-import org.osgi.service.http.HttpContext;
-import org.osgi.service.http.HttpService;
+import org.osgi.service.http.whiteboard.propertytypes.HttpWhiteboardServletName;
+import org.osgi.service.http.whiteboard.propertytypes.HttpWhiteboardServletPattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * A servlet that serves audio streams via HTTP.
  *
  * @author Kai Kreuzer - Initial contribution
  */
+@Component(service = { AudioHTTPServer.class, Servlet.class })
+@HttpWhiteboardServletName(AudioServlet.SERVLET_PATH)
+@HttpWhiteboardServletPattern(AudioServlet.SERVLET_PATH + "/*")
 @NonNullByDefault
```

Comment:
This is more a comment about code-style (and not necessarily to be enforced here): There are dozens of different arrangements for the annotations, sometime `@Component` is first and `@NonNullByDefault` is at the end, sometime it's exactly the opposite and sometime they are together with the whiteboard annotations above or below. 

Can we decide for one convention and probably enable https://github.com/checkstyle/checkstyle/issues/11693 if it is implemented in the customizable way someday? I would agree with https://stackoverflow.com/a/5638388 and order by importance, i.e. `@Component`, other and `@NonNullByDefault` at the end as it is present on nearly every class.

Reply:
I usually sort them which most of the time also goes well with the order of importance.